### ISSUE TITLE
Missing incompatible override error with decorator

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#2838]
**Describe the bug**
No `incompatibleMethodOverride` error is raised if a decorator changes the method signature.

**To Reproduce**
```py
from typing import Callable, ParamSpec

P = ParamSpec("P")

def decorator(
    func: Callable[P, None]
) -> Callable[P, int]:
    ...

class Parent:
    def func(self, var: int) -> None:
        ...

class Child(Parent):
    @decorator
    def func(self, var: int) -> None:
        ...
```

**Expected behavior**
`incompatibleMethodOverride` error for `Child.func`

**VS Code extension or command-line**
Command-line, v1.1.205

**Additional context**
Mypy output
```
error: Return type "int" of "func" incompatible with return type "None" in supertype "Parent"
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.206, which I just published. It will also be in the next release of pylance.
